drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/case_note_export.rb

```diff
@@ -0,0 +1,62 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: false
+
+module HmisExternalApis::AcHmis::Exporters
+  class CaseNoteExport
+    include ::HmisExternalApis::AcHmis::Exporters::CsvExporter
+
+    def run!
+      Rails.logger.info 'Generating content of case note export'
+
+      write_row(columns)
+      total = case_notes.count
+
+      Rails.logger.error "There are #{total} case notes to export. That doesn't look right" if total < 10
+
+      case_notes.find_each.with_index do |case_note, i|
+        Rails.logger.info "Processed #{i} of #{total}" if (i % 1000).zero?
+
+        warehouse_id = case_note.client.warehouse_id
```

Comment:
`case_notes` isn't joining `client` so unless `case_note.enrollment` goes through a client, you can't count on having the client here, and may want:
```suggestion
        warehouse_id = case_note.client&.warehouse_id
```

Reply:
Hm good point. It shouldn't happen (since when we set the enrollment we also set the PersonalID; and because the PersonalID column is non-nullable ) but I'll add the safe accessor just in case!